iron-component-page.html

```diff
@@ -316,28 +317,38 @@
 
     _frameSrc: function(view) {
       if (!view || view.indexOf("demo:") !== 0) return "about:blank";
+
       var src = view.split(':')[1];
-      return new URL(src, this.base).toString();
+      var demoSrc = new URL(src, this.base).toString();
+
+      // If you use history.pushState with iframe.src = url, you will create 2 history entries,
+      // but setting the `src` attribute imperatively will not.
+      if (this.$.demo.src != demoSrc) {
+        this.$.demo.src = demoSrc;
+      }
+    },
+
+    _descriptorsChanged: function() {
+      this.active = this._getDefaultActive();
     },
 
-    _descriptorsChanged: function(docElements, docBehaviors) {
+    _getDefaultActive: function() {
       function findMatch(el) {
         return el && el.contentHref && el.contentHref.indexOf(mainFile) > 0;
       }
 
       var url = this._srcUrl || this.base;
       var mainFile = url.replace(_baseUrl(this.base), '');
-      var matchedPage = docElements.find(findMatch) || docBehaviors.find(findMatch);
+      var matchedPage = this.docElements.find(findMatch) || this.docBehaviors.find(findMatch);
```

Comment:
I set the initial value to an empty array. `docElements` and `docBehaviors` should never be null, so we don't need to check them all the time.